### PR TITLE
3.x into 4.x

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
@jordisala1991 How do I fix the build ?

**Comment by jordisala1991** [comment]
You need to add return types (atleast as a comment, but ideally if it cant be bc as a php return type directly)

Edit: Only to those methods reported on the test as direct deprecations.

File: src/Backend/BackendHealthCheck.php [hunk 1]
```diff
@@ -30,11 +30,17 @@ public function __construct(BackendInterface $backend)
         $this->backend = $backend;
     }
 
+    /**
+     * @return \Laminas\Diagnostics\Result\ResultInterface
```
Comment (jordisala1991) [inline src/Backend/BackendHealthCheck.php]: This can be added on use section

File: src/Iterator/AMQPMessageIterator.php [hunk 2]
```diff
@@ -80,7 +83,7 @@ public function key(): void
         $this->counter;
     }
 
-    public function valid()
+    public function valid(): bool
```
Comment (jordisala1991) [inline src/Iterator/AMQPMessageIterator.php:86]: isnt this a bc break? or the class is final?

Comment (VincentLanglet) [reply src/Iterator/AMQPMessageIterator.php:86]: the class is final

File: src/Backend/BackendHealthCheck.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Backend/BackendHealthCheck.php]: done

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**